### ISSUE TITLE
"NoSleep kernel extension is not loaded"

### ISSUE THREAD

**Issue by GoogleCodeExporter** [irishuang1211/macosx-nosleep-extension#76]
```
What steps will reproduce the problem?
1. Download and Install version 1.3.2
2. Near the end of the install process, "NoSleep kernel extension is not 
loaded" appears (though installation is reported to be successful).
3. Any attempt to start the program (i.e., System Preferences) gives the same 
error. 

What is the expected output? What do you see instead?
Expected the NoSleep icon in menubar, never appears. Error message "NoSleep 
kernel extension is not loaded" appears instead.

What version of the product are you using? On what operating system?
NoSleep 1.3.2, on Mac OS 10.6.8. Downgrading to 1.3.1 works, but was hoping to 
get the bug fixes, since the menubar icon only switches one setting, and 
opening preferences crashes the extension.
```

Original issue reported on code.google.com by `evanwilh...@gmail.com` on 27 Mar 2013 at 5:19

**Comment by GoogleCodeExporter** [comment]
```
Hi,

Could you try the installation again and 

1) Check if the kext is present on the filesystem: 
/System/Library/Extensions/NoSleep.kext

2) If it's actually loaded: kextstat | grep NoSleep
the output should be like this:
  133    0 0xffffff7f81e66000 0x4000     0x4000     com.protech.NoSleep (1.3.2) <5 4 3> 

3) if it's exists but not loaded - try load it manually by: sudo kextutil 
/System/Library/Extensions/NoSleep.kext

4) if something goes wrong during the load, please dump the dmesg log just 
after this by: sudo dmesg

Regards,
--Pavel.
```

Original comment by `integral...@gmail.com` on 27 Mar 2013 at 5:28

**Comment by GoogleCodeExporter** [comment]
Original comment by `integral...@gmail.com` on 27 Mar 2013 at 5:28
- Added labels: **Priority-High**
- Removed labels: **Priority-Medium**

**Comment by GoogleCodeExporter** [comment]
```
Thanks for your help, Pavel!

1) Check if the kext is present on the filesystem: 
/System/Library/Extensions/NoSleep.kext

Yes, it is.

2) If it's actually loaded: kextstat | grep NoSleep
No output from that command.

3) if it's exists but not loaded - try load it manually by: sudo kextutil 
/System/Library/Extensions/NoSleep.kext

I get the following error:
Warnings: 
    Executable does not contain code for architecture: 
        i386

Can't load /System/Library/Extensions/NoSleep.kext - no code for running 
kernel's architecture.
Failed to load /System/Library/Extensions/NoSleep.kext - (libkern/kext) 
requested architecture/executable not found.

4) if something goes wrong during the load, please dump the dmesg log just 
after this by: sudo dmesg

sorId=1 LocalApicId=1 Enabled
calling mpo_policy_init for Quarantine
Security policy loaded: Quarantine policy (Quarantine)
calling mpo_policy_init for Sandbox
Security policy loaded: Seatbelt sandbox policy (Sandbox)
calling mpo_policy_init for TMSafetyNet
Security policy loaded: Safety net for Time Machine (TMSafetyNet)
Copyright (c) 1982, 1986, 1989, 1991, 1993
    The Regents of the University of California. All rights reserved.

MAC Framework successfully initialized
using 9175 buffer headers and 4096 cluster IO buffer headers
IOAPIC: Version 0x11 Vectors 64:87
ACPI: System State [S0 S3 S4 S5] (S3)
PFM64 0xf10000000, 0xf0000000
[ PCI configuration begin ]
console relocated to 0xf10010000
PCI configuration changed (bridge=1 device=1 cardbus=0)
[ PCI configuration end, bridges 3 devices 22 ]
AppleIntelCPUPowerManagement: (built 16:44:45 Jun  7 2011) initialization 
complete
mbinit: done (64 MB memory set for mbuf pool)
rooting via boot-uuid from /chosen: A3EA43A4-84D3-3626-BE62-DC142F1B569D
Waiting on <dict ID="0"><key>IOProviderClass</key><string 
ID="1">IOResources</string><key>IOResourceMatch</key><string 
ID="2">boot-uuid-media</string></dict>
com.apple.AppleFSCompressionTypeZlib kmod start
com.apple.AppleFSCompressionTypeZlib load succeeded
AppleIntelCPUPowerManagementClient: ready
Got boot device = 
IOService:/AppleACPIPlatformExpert/PCI0@0/AppleACPIPCI/SATA@A/AppleMCP89AHCI/PRT
0@0/IOAHCIDevice@0/AppleAHCIDiskDriver/IOAHCIBlockStorageDevice/IOBlockStorageDr
iver/APPLE SSD TS128C Media/IOGUIDPartitionScheme/Customer@2
BSD root: disk0s2, major 14, minor 2
USBMSC Identifier (non-unique): 000000009833 0x5ac 0x8403 0x9833
hfs: Removed 1 orphaned / unlinked files and 0 directories 
AppleUSBMultitouchDriver::checkStatus - received Status Packet, Payload 2: 
device was reinitialized
systemShutdown false
BTCOEXIST default 
wl0: Broadcom BCM4353 802.11 Wireless Controller
5.10.131.42
Previous Shutdown Cause: 3
NVDANV50HAL loaded and registered.
BootCache: hit rate below threshold (915 hits on 1843 lookups)
DSMOS has arrived
AirPort_Brcm43224: Ethernet address 10:93:e9:06:47:c2
IO80211Controller::dataLinkLayerAttachComplete():  adding AppleEFINVRAM 
notification
IO80211Interface::efiNVRAMPublished():  
AirPort: Link Down on en1. Reason 1 (Unspecified).
ACPI_SMC_Idle_CtrlLoop::initPlatformCtrlLoop - p-state count 3 so forced idle 
should be handled by ACPI_SMC_CtrlLoop - exiting
Auth result for: 00:0b:86:5d:1e:8a Auth timed out
Auth result for: 00:1a:1e:9e:b8:c2 MAC AUTH succeeded
AirPort: Link Up on en1
EDID does not contain Extended EDID, not valid for audio
CODE SIGNING: cs_invalid_page(0x1000): p=125[GoogleSoftwareUp] clearing CS_VALID
CODE SIGNING: cs_invalid_page(0x1000): p=151[ksadmin] clearing CS_VALID
CODE SIGNING: cs_invalid_page(0x1000): p=154[ksadmin] clearing CS_VALID
CODE SIGNING: cs_invalid_page(0x1000): p=163[ksadmin] clearing CS_VALID
AppleMCP89TMS::powerGatingDown(0): Done
CODE SIGNING: cs_invalid_page(0x1000): p=185[ksadmin] clearing CS_VALID
CODE SIGNING: cs_invalid_page(0x1000): p=186[ksadmin] clearing CS_VALID
Auth result for: 00:0b:86:5d:1e:8a MAC AUTH succeeded
com_protech_nosleepextension: setting state: 0, for mode: 0 (ac-mode)
com_protech_nosleepextension: successfully started
com_protech_nosleepextension: setting state: 1, for mode: 1 (ac-mode)
com_protech_nosleepextension: setting state: 0, for mode: 0 (ac-mode)
         0 [Time 1364405170] [Message System Sleep
Wake reason = EC.LidOpen
System Wake
AppleMCP89TMS::powerGatingDown(0): Done
Previous Sleep Cause: 5
AirPort: Link Down on en1. Reason 4 (Disassociated due to inactivity).
AppleUSBMultitouchDriver::checkStatus - received Status Packet, Payload 2: 
device was reinitialized
Auth result for: 00:0b:86:5d:1e:8a Auth timed out
Auth result for: 00:1a:1e:9d:89:02 MAC AUTH succeeded
AirPort: Link Up on en1
Auth result for: 00:0b:86:5d:1e:8a MAC AUTH succeeded
CODE SIGNING: cs_invalid_page(0x1000): p=758[GoogleSoftwareUp] clearing CS_VALID
com_protech_nosleepextension: setting state: 0, for mode: 0 (ac-mode)
com_protech_nosleepextension: successfully stopped

```

Original comment by `evanwilh...@gmail.com` on 27 Mar 2013 at 6:26

**Comment by GoogleCodeExporter** [comment]
```
Could you try this special version?
```

Original comment by `integral...@gmail.com` on 27 Mar 2013 at 6:43

**Comment by GoogleCodeExporter** [comment]
```
I have the same symptoms:
Warnings: 
    Executable does not contain code for architecture: 
        i386
```

Original comment by `Alexande...@gmail.com` on 27 Mar 2013 at 6:43

**Comment by GoogleCodeExporter** [comment]
```
Pavel,

That package works! 

Thanks for your help, this might be the most individual attention I've ever 
received in submitting a bug for a piece of software.


```

Original comment by `evanwilh...@gmail.com` on 27 Mar 2013 at 7:36

**Comment by GoogleCodeExporter** [comment]
```
You are welcome:)

I will return i386 support in the next short release.

--P.

```

Original comment by `integral...@gmail.com` on 27 Mar 2013 at 7:40

**Comment by GoogleCodeExporter** [comment]
```
Fixed in 1.3.3
```

Original comment by `integral...@gmail.com` on 27 Mar 2013 at 8:25
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
It's important to reinstall the special version of NoSleep from this thread to 
regular 1.3.3 from Downloads section. 

The autoupdate could be broken in this version.
```

Original comment by `integral...@gmail.com` on 27 Mar 2013 at 8:29
